tests/test_database_types.py

```diff
@@ -0,0 +1,222 @@
+import unittest
+import preql
+import time
+from data_diff import database as db
+from data_diff.diff_tables import TableDiffer, TableSegment, split_space
+from parameterized import parameterized, parameterized_class
+from .common import CONN_STRINGS, str_to_checksum
+import logging
+
+logging.getLogger("diff_tables").setLevel(logging.WARN)
+logging.getLogger("database").setLevel(logging.WARN)
+
+CONNS = {k: db.connect_to_uri(v) for k, v in CONN_STRINGS.items()}
+
+TYPE_SAMPLES = {
+    "int": [127, -3, -9, 37, 15, 127],
+    "datetime_no_timezone": [
+        "2020-01-01 15:10:10",
+        "2020-01-01 9:9:9",
+        "2022-01-01 15:10:01.139",
+        "2022-01-01 15:10:02.020409",
+        "2022-01-01 15:10:03.003030",
+        "2022-01-01 15:10:05.009900",
```

Comment:
I think we should probably make some of these 9 precision, just for a bit of extra testing.

Reply:
Can't get it done with the way our tests work right now, using Python's Datetime to copy the values from source to target. I can't see a simple solution to it, other than hardcode this scenario.